lib/workers/global/autodiscover.ts

```diff
@@ -95,10 +100,13 @@ export async function autodiscoverRepositories(
   return { ...config, repositories: discovered };
 }
 
-export function applyFilters(repos: string[], filters: string[]): string[] {
-  const matched = new Set<string>();
-
-  for (const filter of filters) {
+export function applyFilters(
+  repos: string[],
+  addFilters: string[],
+  removeFilters: string[],
+): string[] {
+  const applyFilter = (repos: string[], filter: string): string[] => {
```

Comment:
Do not use local functions 

https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md#functions